src/googmodule.ts

```diff
@@ -0,0 +1,388 @@
+        const ident: ts.Identifier|undefined = namespaceToModuleIdentifier.get(imp.namespace.text);
+        let initializer: ts.Expression;
+        if (!ident) {
+          namespaceToModuleIdentifier.set(imp.namespace.text, newIdent);
+          initializer = createGoogCall('require', imp.namespace);
+        } else {
+          initializer = ident;
+        }
+        const varDecl = ts.createVariableDeclaration(newIdent, /* type */ undefined, initializer);
+        const newStmt = ts.createVariableStatement(
+            /* modifiers */ undefined, ts.createVariableDeclarationList([varDecl]));
+        return ts.setOriginalNode(ts.setTextRange(newStmt, original), original);
+      }
+
+      function visitTopLevelStatement(
+          statements: ts.Statement[], sf: ts.SourceFile, node: ts.Statement): void {
+        switch (node.kind) {
+          case ts.SyntaxKind.ExpressionStatement: {
+            const exprStmt = node as ts.ExpressionStatement;
+            // Check for "use strict" and certain Object.defineProperty and skip it if necessary.
+            if (isUseStrict(exprStmt) || isEsModuleProperty(exprStmt)) {
+              stmts.push(ts.createNotEmittedStatement(exprStmt));
+              return;
+            }
+            // Check for:
+            //   module.exports = ...;
+            const modExports = rewriteModuleExportsAssignment(exprStmt);
+            if (modExports) {
+              stmts.push(modExports);
+              return;
+            }
+            // Check for:
+            //   "require('foo');" (a require for its side effects)
+            const expr = exprStmt.expression;
+            if (!ts.isCallExpression(expr)) break;
+            let callExpr = expr;
+            // Handle export * in ES5 mode (in ES6 mode, export * is dereferenced already).
+            // export * creates either a pure top-level '__exportStar(require(...))' or the imported
+            // version, 'tslib.__exportStar(require(...))'. The imported version is only substituted
+            // later on though, so appears as a plain "__exportStar" on the top level here.
+            const isExportStar =
+                (ts.isIdentifier(expr.expression) && expr.expression.text === '__exportStar') ||
+                (ts.isIdentifier(expr.expression) && expr.expression.text === '__export');
+            if (isExportStar) callExpr = expr.arguments[0] as ts.CallExpression;
+            const ident = ts.createIdentifier(nextModuleVar());
+            const require = maybeCreateGoogRequire(exprStmt, callExpr, ident);
+            if (!require) break;
+            statements.push(require);
+            // If this is an export star
+            if (isExportStar) {
+              const args: ts.Expression[] = [ident];
+              if (expr.arguments.length > 1) args.push(expr.arguments[1]);
+              statements.push(ts.createStatement(ts.createCall(expr.expression, undefined, args)));
+            }
+            return;
```

Comment:
The flow through this is a bit weird.  It seems like it's a series of
if (special case) handleIt && return;
and then "break" means "do the normal processing".  Here, I would expect the isExportStar to return and then it to break after.

If that's not the case, maybe some comments about what is expected to fall out at the end here would help?

Reply:
I've added a comment to the function.

The invariant is that the function will add a node to statements in any case, even if not handling. We need it like that because there is one case where we have to add two nodes (`export *`).

We return once we're done with the node, otherwise break to the default handling. Does that make sense?